provider/cmd/pulumi-resource-aws-apigateway/provider.ts

```diff
@@ -43,6 +43,10 @@ async function constructRestAPI(name: string, inputs: pulumi.Inputs,
         urn: restAPI.urn,
         state: {
             url: restAPI.url,
+            api: restAPI.api,
+            deployment: restAPI,
+            stage: restAPI.stage,
+            apiPolicy: restAPI.apiPolicy,
```

Comment:
As a best practice, it'd be good to register these as outputs of the component as well here:

https://github.com/pulumi/pulumi-aws-apigateway/blob/10e72b20025ed031779db06689bd46cabc3f6e07/provider/cmd/pulumi-resource-aws-apigateway/restAPI.ts#L158-L160